docs/secure-communication-agents.asciidoc

```diff
@@ -156,14 +155,83 @@ The following example invalidates the `java-001` API key.
 
 The response will look similar to this:
 
-[source,console-result,subs="attributes,callouts"]
+[source,console-result]
 --------------------------------------------------
 Invalidated keys ... qT4tz28B1g59zC3uAXfW
 Error count ........ 0
 --------------------------------------------------
 
 A full list of `apikey` subcommands and flags is available in the <<apikey-command,API key command reference>>.
 
+[[create-api-key-workflow-es]]
+[float]
+==== Alternate API key workflow example
+
+Instead of using the APM Server CLI, it is possible to create API keys using the Elasticsearch
+{ref}/security-api-create-api-key.html[create API key API].
+
+This example creates an API key named `java-002`:
+
+[source,kibana]
+----
+POST /_security/api_key
+{
+  "name": "java-002", <1>
+  "expiration": "1d", <2>
+  "role_descriptors": {
+    "apm": {
+      "applications": [
+        {
+          "application": "apm",
+          "privileges": ["sourcemap:write", "event:write", "config_agent:read"], <3>
+          "resources": ["*"]
+        }
+      ]
+    }
+  }
+}
+----
+<1> The name of the API key
+<2> The expiration time of the API key
+<3> Any assigned privileges
+
+The response will look similar to this:
+
+[source,console-result]
+----
+{
+  "id" : "GnrUT3QB7yZbSNxKET6d",
+  "name" : "java-002",
+  "expiration" : 1599153532262,
+  "api_key" : "RhHKisTmQ1aPCHC_TPwOvw"
+}
+----
+
+The `credential` string, which is what agents use to communicate with APM Server,
+is a base64 encoded representation of the API key's `id:api_key`.
+It can be created like this:
+
+[source,console-result]
+--------------------------------------------------
+echo -n GnrUT3QB7yZbSNxKET6d:RhHKisTmQ1aPCHC_TPwOvw | base64
+--------------------------------------------------
+
+Using the returned string, you can verify the API key has the expected privileges with the APM Server CLI:
```

Comment:
Maybe also provide an example of using https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-authenticate.html to check the credentials are correct? That will at least help ensure they base64-encoded it correctly.

Reply:
Thanks, I didn't know about this step. Will do!

Reply:
Is this possible? I can't find any docs that suggest you can do this.

Reply:
Ohh, it's probably because I'm using ESS. Something like is what you're suggesting?

```
curl -H "Authorization: ApiKey R0gzRWIzUUI3eVpiU054S3pYSy06bXQyQWl4TlZUeEcyUjd4cUZDS0NlUQ==" https://ess.io:9243/_security/_authentication
```

Reply:
It works in ESS, I just tried it. If you tried it and it didn't work, maybe you made the same mistake I did to start with and sent the request to the Kibana URL instead of Elasticsearch.

Anyway, yes, what you pasted there is what I'm suggesting.